### ISSUE TITLE
Issue related to table processing

### ISSUE THREAD

**Issue by kamyuentse** [jrblevin/markdown-mode#308]
## Expected Behavior
Something we may use `|` inside the cell, and use `\|` to represent it.

For example, when typing then press `Enter`
```
| Col1  | Col2  |
| :-: | :-: |
|  AAA  |  A\|B |
```

I expect to get 

```
| Col1  | Col2  |
| :-:   | :-:   |
|  AAA  |  A\|B |
```

## Actual Behavior

But currently, we get

```
| Col1  | Col2  |   |
| :---: | :---: |   |
| AAA   | A\    | B |
|       |       |   |
```

Maybe this is related to the `hline` matching rule?
## Steps to Reproduce

1. Step A
2. Step B
3. Step C

## Software Versions

- Markdown Mode: 
- Emacs: 
- OS:

**Comment by saf-dmitry** [comment]
Escaping of pipe symbols inside tables is currently not implemented.

You can try to include pipe symbol in cells using the ASCII HTML character code `&#124;` or named entity `&vert;`. Results may depend on your Markdown implementation and output format. In inline LaTeX math statements this problem can be avoided by using the LaTeX `\vert` command instead of `|`.

P.S. For proper detection of horizontal separator lines _no padding_ is allowed between the beginning pipe character `|` and header separator symbol `:` or `-`.

**Comment by kamyuentse** [comment]
@saf-dmitry Thanks a lot, and I have another problem, how can I fold/unfold the HTML elements and control blocks in markdown files.

**Comment by dankessler** [comment]
A similar issue also occurs if a Wiki-style link exists in a table cell.

For example, I can start with this table:
```
| A           | B       |
|-------------|---------|
| [[Address]] | Content |
```
and then add a `|` to the link, for Wiki-style linking
```
| A           | B       |
|-------------|---------|
| [[Page|Address]] | Content |
```

and after hitting tab (with the goal of fixing alignment), I get
```
| A      | B         |         |
|--------|-----------|---------|
| [[Page | Address]] | Content |
```
where we see that the pipe has been interpreted as a column break, and the contents of the link are now split across cells, and a new column has appeared.

This is similar to the behavior described above, but AFAIK there isn't a work-around to avoid the use of the pipe in the wiki style link.

**Comment by dankessler** [comment]
My primary goal of this was to get things to render nicely in github wiki's, and I've since discovered that the use of wiki links inside of table cells seems to break things in much the same that it confuses `markdown-mode`, so maybe this is not quite the bug I thought it was...

**Comment by syohex** [comment]
I have merged #475. I suppose that it fixes both escaped `|` and wiki-link issues. Please check latest version.
